src/sign/records.rs

```diff
@@ -492,11 +492,10 @@ impl<'a, N, D> RecordsIter<'a, N, D> {
     where
         N: ToName,
     {
-        while let Some(first) = self.slice.first() {
-            if first.class() != apex.class() {
-                continue;
-            }
-            if apex == first || first.owner().ends_with(apex.owner()) {
+        for rr in self.slice {
```

Comment:
For-looping on a thing that you are also currently modifying sounds scary and unintuitive. I’m kind of surprised the compiler lets you do this.

Reply:
In my opinion, signing needs to be changed to not allow multiple classes. There is no need to that feature and it just make the code more complex.